tests/tool_use/utils.py

```diff
@@ -7,9 +9,56 @@
 from tests.utils import VLLM_PATH
 
 
-class ServerConfig(TypedDict):
+class ServerConfig(TypedDict, total=False):
     model: str
     arguments: List[str]
+    system_prompt: Optional[str]
+    supports_parallel: Optional[bool]
+    format_tool_output: Optional[Callable[[str], str]]
+
+
+def format_llama_tool_output(output: str) -> str:
+    return json.dumps({"output": output})
+
+
+def format_tool_output_id(output: str) -> str:
+    return output
+
+
+def patch_tool_output(messages: List[Dict[str, Any]],
+                      config: ServerConfig) -> List[Dict[str, Any]]:
+    fmt_fun = config.get("format_tool_output")
+    if not fmt_fun:
+        return messages
+    new_messages = deepcopy(messages)
+    for message in new_messages:
+        if message["role"] == "tool":
+            message["content"] = fmt_fun(message["content"])
+    return new_messages
+
+
+def patch_system_prompt(messages: List[Dict[str, Any]],
+                        system_prompt: str) -> List[Dict[str, Any]]:
+    new_messages = deepcopy(messages)
+    if new_messages[0]["role"] == "system":
+        new_messages[0]["content"] = system_prompt
+    else:
+        new_messages.insert(0, {"role": "system", "content": system_prompt})
+    return new_messages
+
+
+def ensure_system_prompt(messages: List[Dict[str, Any]],
+                         config: ServerConfig) -> List[Dict[str, Any]]:
+    prompt = config.get("system_prompt")
+    if prompt:
+        return patch_system_prompt(messages, prompt)
+    else:
+        return messages
+
+
+def adapt_prompt_to_model(messages: List[Dict[str, Any]],
+                          config: ServerConfig) -> List[Dict[str, Any]]:
+    return ensure_system_prompt(patch_tool_output(messages, config), config)
```

Comment:
This seems kind of like overkill. If a model needs a specific system prompt to get tools working (both Mistral and hermes do for best results) that should be in the chat template provided in `examples/` since the goal of the tool chat templates in "examples" is to have something that just works out of the box.

Reply:
I agree that it makes sense to add to the chat template. But during the tests the example chat template is not the one that is used but rather the one that comes with the model in the `tokenizer_config.json` file.

Reply:
It seems like you have the tests correctly set up to use the example chat template - from `utils.py`:

```
"llama": {
        "model":
        "meta-llama/Meta-Llama-3.1-8B-Instruct",
        "arguments": [
            "--tool-call-parser", "llama", "--chat-template",
            str(VLLM_PATH / "examples/tool_chat_template_llama.jinja")
        ],
```